paginate-nfts/README.md

```diff
@@ -0,0 +1,90 @@
+# Paginating NFTs for a given wallet address
+
+Pagination is not really a thing when it comes to `getProgramAcccount` RPC method. This tutorial demonstrates how you can still implement pagination using the [Metaplex JS SDK](https://github.com/metaplex-foundation/js).
+
+We will be using the [findAllByOwner](https://github.com/metaplex-foundation/js#findallbyowner) method in the Metaplex SDK which will give us the list of NFTs without their metadata JSON loaded. (This is crucial because if we have to fetch 1000 NFTs with their metadata, it will take a lot of time and hugely impact the performance of application and will fetch a large amount data in a single go).
+
+This example uses a new NextJS app with Metaplex which can be created by following the instructions listed in [Getting Started with Metaplex and Next.js](../getting-started-nextjs).
+
+In this tutorial, we are going to:
+- Use the `findAllByOwner` method
+- Store the list of lazy-loaded NFTs
+- Fetch the metadata of for each NFT in the current page
+
+![image](https://user-images.githubusercontent.com/56197821/176737427-43df0aca-6ea9-443f-b9e9-718bdb654ab4.png)
+
+Okay, let's get started 🔥 ( You can find all the code in this [file](./pages/index.js))
+
+1. **`fetchNFTs` Function is called**
+   
+   Once the user fills in the wallet address and clicks Fetch button, `fetchNFTs` function is called.
+
+   ```js
+   const fetchNFTs = async () => {
+      try {
+         setLoading(true);
+         setCurrentView(null);
+         const list = await mx.nfts().findAllByOwner(new PublicKey(address));
+         setNftList(list);
+         setCurrentPage(1);
+      } catch (e) {
+         console.error(e);
+      }
+   };
+   ```
+   This sets the `Loading` state to true, invalidates the current view and makes a call to `findAllByOwner` method with wallet address as input.
+
+   The data is set to `nftList` state and current page is set to default 1 (first page).
+
+2. **UseEffect gets Triggered**
+   
+   Once the nftList is in place and currentPage is set to 1, it triggers the useEffect.
+
+   ```js
+   useEffect(() => {
+   if (nftList) {
+   const execute = async () => {
+   const startIndex = (currentPage - 1) * perPage;
+   const endIndex = currentPage * perPage;
+   await loadData(startIndex, endIndex);
+   setCurrentView(nftList.slice(startIndex, endIndex));
+   setLoading(false);
+   };
+   execute();
+   }
+   }, [nftList, currentPage]);
```

Comment:
Fixed bad indentation and returned early so if the nftList is null to make it easier to read. Please make sure this is also updated in the index.js file.

```suggestion
   useEffect(() => {
     if (!nftList) {
       return;
     }
     
     const execute = async () => {
       const startIndex = (currentPage - 1) * perPage;
       const endIndex = currentPage * perPage;
       await loadData(startIndex, endIndex);
       setCurrentView(nftList.slice(startIndex, endIndex));
       setLoading(false);
     };
     
     execute();
   }, [nftList, currentPage]);
```